dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/config/DubboConfigEarlyInitializationPostProcessor.java

```diff
@@ -45,27 +42,27 @@
  * @see GenericBeanPostProcessorAdapter
  * @since 2.7.9
  */
-public class DubboConfigEarlyInitializationPostProcessor extends GenericBeanPostProcessorAdapter<AbstractConfig>
-        implements BeanDefinitionRegistryPostProcessor, PriorityOrdered {
+public class DubboConfigEarlyInitializationPostProcessor extends GenericBeanPostProcessorAdapter<AbstractConfig> {
 
     private static final Log logger = LogFactory.getLog(DubboConfigEarlyInitializationPostProcessor.class.getName());
 
-    public static final String BEAN_NAME = "dubboConfigEarlyInitializationPostProcessor";
+    private static DubboConfigEarlyInitializationPostProcessor SINGLETON;
 
     private DefaultListableBeanFactory beanFactory;
 
-    @Override
-    public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry) throws BeansException {
-        this.beanFactory = unwrap(registry);
-        initBeanFactory();
+    private DubboConfigEarlyInitializationPostProcessor(DefaultListableBeanFactory beanFactory) {
+        this.beanFactory = beanFactory;
     }
 
-    @Override
-    public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
-        if (beanFactory == null) { // try again if postProcessBeanDefinitionRegistry method does not effect.
-            this.beanFactory = unwrap(beanFactory);
-            initBeanFactory();
+    public static DubboConfigEarlyInitializationPostProcessor getSingleton(DefaultListableBeanFactory beanFactory) {
```

Comment:
The global instance of DubboConfigEarlyInitializationPostProcessor  holds in DubboConfigEarlyRegistrationPostProcessor is better.  
DubboConfigEarlyInitializationPostProcessor  just a  BeanPostProcessor , it should not care for how to create and be registered.

Reply:
I think the DubboConfigEarlyInitializationPostProcessor can be changed to a inner class of DubboConfigEarlyRegistrationPostProcessor,  it's managed by DubboConfigEarlyRegistrationPostProcessor.